components/CopyButton.tsx

```diff
@@ -2,17 +2,55 @@
 
 import { Button } from "@/components/ui/button";
 import { toast } from "sonner";
+import { useState, useEffect } from "react";
+import { Check } from "lucide-react";
 
 export default function CopyButton({ label }: { label: string }) {
+  const [copied, setCopied] = useState(false);
+  const [loaded, setLoaded] = useState(false);
+
+  useEffect(() => {
+    const already_copied = localStorage.getItem("already_copied");
```

Comment:
camelCase for JS and personal preference
```suggestion
    const isAlreadyCopied = Boolean(localStorage.getItem("already_copied"));
```

Reply:
`already_copied` is actually not a boolean here, it is a JSON object.

we do the `if (!already_copied)` check because when doing `localStorage.getItem()` the return type is `string | null` (null when the key hasn't been set yet, which is why we set it to a `JSON.stringify`d empty object.

good call on the camelCase, will adjust

Reply:
true but i don't see us using it as a JSON object just checking for it's existence - hence the Boolean conversion

Reply:
https://github.com/KMontag42/afk-journey-team-builder/pull/44/files#diff-80937245eeb9f8a734bd2c051b2014ab6ca476d4086b1ee8e85036a5deaaa40dR17